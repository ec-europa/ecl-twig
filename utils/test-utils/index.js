const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
