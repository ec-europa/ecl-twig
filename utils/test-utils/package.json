{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
