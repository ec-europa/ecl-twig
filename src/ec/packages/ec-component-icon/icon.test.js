import path from 'path';
import { renderTwigFile } from '@ecl-twig/test-utils';

import brandedIcons from '@ecl/ec-resources-icons/dist/lists/branded.json';
import generalIcons from '@ecl/ec-resources-icons/dist/lists/general.json';
import notificationsIcons from '@ecl/ec-resources-icons/dist/lists/notifications.json';
import uiIcons from '@ecl/ec-resources-icons/dist/lists/ui.json';

describe('EC - Icon', () => {
  const template = path.resolve(__dirname, './icon.html.twig');
  const defaultPath = 'static/icons.svg';
  const defaultDataStructure = {
    icon: {
      name: '',
      type: '',
      path: defaultPath,
    },
  };

  describe('Branded', () => {
    brandedIcons.forEach(icon => {
      test(`- icon ${icon} renders correctly`, done => {
        expect.assertions(1);

        defaultDataStructure.icon.name = icon;
        defaultDataStructure.icon.type = 'branded';

        renderTwigFile(template, defaultDataStructure, (err, html) => {
          expect(html).toMatchSnapshot();
          done();
        });
      });
    });
  });

  describe('Notifications', () => {
    notificationsIcons.forEach(icon => {
      test(`- icon ${icon} renders correctly`, done => {
        expect.assertions(1);

        defaultDataStructure.icon.name = icon;
        defaultDataStructure.icon.type = 'notifications';

        renderTwigFile(template, defaultDataStructure, (err, html) => {
          expect(html).toMatchSnapshot();
          done();
        });
      });
    });
  });

  describe('General', () => {
    generalIcons.forEach(icon => {
      test(`- icon ${icon} renders correctly`, done => {
        expect.assertions(1);

        defaultDataStructure.icon.name = icon;
        defaultDataStructure.icon.type = 'general';

        renderTwigFile(template, defaultDataStructure, (err, html) => {
          expect(html).toMatchSnapshot();
          done();
        });
      });
    });
  });

  describe('UI', () => {
    uiIcons.forEach(icon => {
      test(`- icon ${icon} renders correctly`, done => {
        expect.assertions(1);

        defaultDataStructure.icon.name = icon;
        defaultDataStructure.icon.type = 'ui';

        renderTwigFile(template, defaultDataStructure, (err, html) => {
          expect(html).toMatchSnapshot();
          done();
        });
      });
    });
  });
});
