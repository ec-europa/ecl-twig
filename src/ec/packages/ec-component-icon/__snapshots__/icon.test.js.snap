// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Icon Branded - icon facebook renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#branded--facebook"
  />
</svg>
`;

exports[`EC - Icon Branded - icon instagram renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#branded--instagram"
  />
</svg>
`;

exports[`EC - Icon Branded - icon linkedin renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#branded--linkedin"
  />
</svg>
`;

exports[`EC - Icon Branded - icon pinterest renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#branded--pinterest"
  />
</svg>
`;

exports[`EC - Icon Branded - icon rss renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#branded--rss"
  />
</svg>
`;

exports[`EC - Icon Branded - icon skype renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#branded--skype"
  />
</svg>
`;

exports[`EC - Icon Branded - icon twitter renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#branded--twitter"
  />
</svg>
`;

exports[`EC - Icon Branded - icon youtube renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#branded--youtube"
  />
</svg>
`;

exports[`EC - Icon General - icon audio renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--audio"
  />
</svg>
`;

exports[`EC - Icon General - icon book renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--book"
  />
</svg>
`;

exports[`EC - Icon General - icon calendar renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--calendar"
  />
</svg>
`;

exports[`EC - Icon General - icon copy renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--copy"
  />
</svg>
`;

exports[`EC - Icon General - icon digital renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--digital"
  />
</svg>
`;

exports[`EC - Icon General - icon edit renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--edit"
  />
</svg>
`;

exports[`EC - Icon General - icon faq renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--faq"
  />
</svg>
`;

exports[`EC - Icon General - icon feedback renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--feedback"
  />
</svg>
`;

exports[`EC - Icon General - icon file renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--file"
  />
</svg>
`;

exports[`EC - Icon General - icon gear renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--gear"
  />
</svg>
`;

exports[`EC - Icon General - icon generic-lang renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--generic-lang"
  />
</svg>
`;

exports[`EC - Icon General - icon global renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--global"
  />
</svg>
`;

exports[`EC - Icon General - icon growth renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--growth"
  />
</svg>
`;

exports[`EC - Icon General - icon image renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--image"
  />
</svg>
`;

exports[`EC - Icon General - icon infographic renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--infographic"
  />
</svg>
`;

exports[`EC - Icon General - icon language renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--language"
  />
</svg>
`;

exports[`EC - Icon General - icon livestreaming renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--livestreaming"
  />
</svg>
`;

exports[`EC - Icon General - icon location renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--location"
  />
</svg>
`;

exports[`EC - Icon General - icon multiple-files renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--multiple-files"
  />
</svg>
`;

exports[`EC - Icon General - icon organigram renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--organigram"
  />
</svg>
`;

exports[`EC - Icon General - icon print renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--print"
  />
</svg>
`;

exports[`EC - Icon General - icon regulation renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--regulation"
  />
</svg>
`;

exports[`EC - Icon General - icon search renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--search"
  />
</svg>
`;

exports[`EC - Icon General - icon share renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--share"
  />
</svg>
`;

exports[`EC - Icon General - icon spinner renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--spinner"
  />
</svg>
`;

exports[`EC - Icon General - icon spreadsheet renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--spreadsheet"
  />
</svg>
`;

exports[`EC - Icon General - icon video renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--video"
  />
</svg>
`;

exports[`EC - Icon Generic tests - Any icon renders correctly with extra attributes 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  data-test="data-test-value"
  data-test-1="data-test-value-1"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--audio"
  />
</svg>
`;

exports[`EC - Icon Generic tests - Any icon renders correctly with extra class names 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m custom-class custom-class--test"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#general--audio"
  />
</svg>
`;

exports[`EC - Icon Notifications - icon error renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#notifications--error"
  />
</svg>
`;

exports[`EC - Icon Notifications - icon information renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#notifications--information"
  />
</svg>
`;

exports[`EC - Icon Notifications - icon success renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#notifications--success"
  />
</svg>
`;

exports[`EC - Icon Notifications - icon warning renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#notifications--warning"
  />
</svg>
`;

exports[`EC - Icon UI - icon check renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#ui--check"
  />
</svg>
`;

exports[`EC - Icon UI - icon check-filled renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#ui--check-filled"
  />
</svg>
`;

exports[`EC - Icon UI - icon close renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#ui--close"
  />
</svg>
`;

exports[`EC - Icon UI - icon close-filled renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#ui--close-filled"
  />
</svg>
`;

exports[`EC - Icon UI - icon corner-arrow renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#ui--corner-arrow"
  />
</svg>
`;

exports[`EC - Icon UI - icon download renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#ui--download"
  />
</svg>
`;

exports[`EC - Icon UI - icon external renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#ui--external"
  />
</svg>
`;

exports[`EC - Icon UI - icon fullscreen renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#ui--fullscreen"
  />
</svg>
`;

exports[`EC - Icon UI - icon minus renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#ui--minus"
  />
</svg>
`;

exports[`EC - Icon UI - icon plus renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#ui--plus"
  />
</svg>
`;

exports[`EC - Icon UI - icon rounded-arrow renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#ui--rounded-arrow"
  />
</svg>
`;

exports[`EC - Icon UI - icon solid-arrow renders correctly 1`] = `
<svg
  aria-hidden="true"
  class="ecl-icon ecl-icon--m"
  focusable="false"
>
  <use
    xlink:href="static/icons.svg#ui--solid-arrow"
  />
</svg>
`;
